lib/navigation/preprocessing/accelerometer_trajectory.cpp

```diff
@@ -23,24 +23,13 @@ void AccelerometerTrajectoryEstimator::update(const core::Float acceleration,
   const auto time_elapsed                = timer_.elapsed(timestamp, previous_timestamp_);
   const core::Float time_elapsed_seconds = timer_.elapsedTimeInSeconds(time_elapsed);
 
-  // from equation v=u+at
-  const core::Float velocity_estimate = velocity_estimate_ + (acceleration * time_elapsed_seconds);
-
-  // from equation s = ut + 0.5*a*(t^2)
-  displacement_estimate_ = (velocity_estimate_ * time_elapsed_seconds)
+  // from equation s = s_0 + ut + 0.5*a*(t^2)
+  displacement_estimate_ = displacement_estimate_ + (velocity_estimate_ * time_elapsed_seconds)
                            + (0.5 * acceleration * time_elapsed_seconds * time_elapsed_seconds);
-  velocity_estimate_  = velocity_estimate;
-  previous_timestamp_ = timestamp;
-}
-
-core::Float AccelerometerTrajectoryEstimator::getDisplacementEstimate() const
-{
-  return displacement_estimate_;
-}
 
-core::Float AccelerometerTrajectoryEstimator::getVelocityEstimate() const
-{
-  return velocity_estimate_;
+  // from equation v=u+at
+  velocity_estimate_  = velocity_estimate_ + (acceleration * time_elapsed_seconds);
```

Comment:
should this be updated before or after the displacement estimate, as displacement uses this value

Reply:
It would make sense before, since the u in displacement is the initial velocity. I changed it in order not to use temp variable, if you look at previous code you'll see that it does same thing.

Reply:
Oh my bad, I should have noticed that. All good then